1. We created the asyncWrapper middleware function in order to streamline our code - the asyncWrapper function creates the
try/catch block in one place rather than repeating it for each controller function.

2. 
const getTask = asyncWrapper(async (req, res, next) => {
  const { id: taskID } = req.params;
  const task = await Task.findOne({ _id: taskID });
  if (!task) {
    return next(createCustomError(`No task with id: ${taskID}`, 404));
  }
  res.status(200).json({ task });
});